pkgs/tools/typesetting/tex/texlive/default.nix

```
@@ -39,6 +50,163 @@ let
+      };
+
+      pedigree-perl = orig.pedigree-perl // {
+        postFixup = ''
+          sed -e '1{ /perl$/!q1 ; s|perl$|perl -I '"$run"'/scripts/pedigree-perl| } ' -i "$out"/bin/*
+        '';
+      };
+
+      pdfcrop = orig.pdfcrop // lib.optionalAttrs
+      (lib.elem "rpdfcrop" orig.pdfcrop.binfiles)
+      {
+        binfiles = lib.remove "rpdfcrop" orig.pdfcrop.binfiles;
+        postFixup = ''
+          cp "$out"/bin/pdfcrop "$out"/bin/rpdfcrop
+        '';
+      };
+
+      pdftex = orig.pdftex // {
+        scriptsFolder = "simpdftex";
+      };
+
+      ptex = orig.ptex // lib.optionalAttrs
+      (lib.all (e: lib.elem e orig.ptex.binfiles) [ "pdvitomp" "pmpost" "r-pmpost" ])
+      {
+        # metapost binaries are in bin.metapost instead of bin.core
+        binfiles = lib.subtractLists [ "pdvitomp" "pmpost" "r-pmpost" ] orig.ptex.binfiles;
+        postFixup = ''
+          ln -t "$out"/bin -s '${bin.metapost}/bin/'{pdvitomp,pmpost,r-pmpost}
+        '';
+      };
+
+      texdef = orig.texdef // lib.optionalAttrs
+      (lib.elem "latexdef" orig.texdef.binfiles)
+      {
+        binfiles = lib.remove "latexdef" orig.texdef.binfiles;
+        postFixup = ''
+          cp "$out"/bin/texdef "$out"/bin/latexdef
+        '';
+      };
+
+      texlive-scripts = orig.texlive-scripts // lib.optionalAttrs
+      (lib.all (e: lib.elem e orig.texlive-scripts.binfiles) [ "man" "mktexfmt" "texhash" ])
+      {
+        scriptsFolder = "texlive";
+        # man should of course be ignored
+        # mktexlsr.pl is better behaved than mktexlsr of texlive.infra
+        binfiles = (lib.subtractLists [ "man" "mktexfmt" "texhash" ] orig.texlive-scripts.binfiles)
+          ++ [ "mktexlsr" ];
+        # fmtutil, updmap need modified perl @INC, including TeXLive::TLUtils from texlive.infra
+        # texhash is a link to mktexlsr and does not check $0 (see texlive/linked_scripts/Makefile.am of TeX Live source)
+        postFixup = ''
+          sed -e '1{ /perl$/!q1 ; s|perl$|perl -I '"$run"'/scripts/texlive -I ${lib.head tl."texlive.infra".pkgs}/tlpkg| } ' -i "$out"/bin/fmtutil "$out"/bin/updmap
+          cp "$out"/bin/fmtutil "$out"/bin/mktexfmt
+          ln -s "$out"/bin/mktexlsr "$out"/bin/texhash
+        '';
+      };
+
+      texlive-scripts-extra = orig.texlive-scripts-extra // lib.optionalAttrs
+      (lib.all (e: lib.elem e orig.texlive-scripts-extra.binfiles) [ "allec" "kpsepath" "kpsexpand" ])
+      {
+        scriptsFolder = "texlive-extra";
+        binfiles = lib.subtractLists [ "allec" "kpsepath" "kpsexpand" "man" ] orig.texlive-scripts-extra.binfiles;
+        # Patch texlinks.sh back to 2015 version;
+        # otherwise some bin/ links break, e.g. xe(la)tex.
+        postFixup = ''
+          patch --verbose -R "$out"/bin/texlinks < '${./texlinks.diff}'
+          ln -s "$out"/bin/allcm "$out"/bin/allec
+          ln -s "$out"/bin/kpsetool "$out"/bin/kspepath
+          ln -s "$out"/bin/kpsetool "$out"/bin/kspexpand
```

```suggestion
          ln -s "$out"/bin/kpsetool "$out"/bin/kpsepath
          ln -s "$out"/bin/kpsetool "$out"/bin/kseexpand
```
These also need to be copies unless the behaviour of the wrapper in `combine` is changed